File: javascript/1. Two Sum/solution.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {


    const map = new Map(); // map for O(1) time retrieval each call

    for (let i = 0; i < nums.length; i++){
        

        if (map.has(target - nums[i])){ // target - nums[i] = the remainder number for selected iteration of numbers
            return [map.get(target - nums[i]), i]; // using "(" instead of "[" creates data type issues
        }

        map.set(nums[i], i); // add an element to map
    }
    return;
};

File: python/1. Two Sum/solution.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        
        map = {}

        for i in range(len(nums)):

            if (target - nums[i] in map):
                return (map[target - nums[i]], i)

            map[nums[i]] = []
            map[nums[i]] = i
        
        return
